core/errors.py

```diff
@@ -2,4 +2,7 @@ class InsufficientStrategiesError(Exception):
     pass
 
 class MissingExportFunctionError(Exception):
-    pass
\ No newline at end of file
+    pass
+
+class TokenPlacementError(Exception):
+    pass
```

Comment:
Nice to see some custom errors, makes it all so much nicer when it something goes to shit.